WordPress/gutenberg
# Shortcut: Covert component to TypeScript

## What?
Converts the `Shortcut` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactors the current `Shortcut` to TypeScript.
- Refactors the test to TypeScript and refactored to use `@testing-library/react`.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Test within the Gutenberg editor
